homeassistant/components/nextdns/__init__.py

```diff
@@ -129,14 +132,27 @@ async def _async_update_data(self) -> AnalyticsProtocols:
             raise UpdateFailed(err) from err
 
 
+class NextDnsSettingsUpdateCoordinator(NextDnsUpdateCoordinator):
```

Comment:
I think that you should set the Generic type on the coordinator:
```suggestion
class NextDnsSettingsUpdateCoordinator(NextDnsUpdateCoordinator[Settings]):
```

And probably also on all the previous ones:
```python
class NextDnsUpdateCoordinator(DataUpdateCoordinator[NextDnsData]):
    ...

class NextDnsStatusUpdateCoordinator(NextDnsUpdateCoordinator[AnalyticsStatus]):
    ...
```

Reply:
But this gives me an exception:
```
File "/home/maciek/develop/home-assistant-core/homeassistant/components/nextdns/__init__.py", line 76, in <module>
    NextDnsUpdateCoordinator[NextDnsUpdateCoordinator]
  File "/usr/local/lib/python3.10/typing.py", line 312, in inner
    return func(*args, **kwds)
  File "/usr/local/lib/python3.10/typing.py", line 1344, in __class_getitem__
    _check_generic(cls, params, len(cls.__parameters__))
  File "/home/maciek/develop/home-assistant-core/venv/lib/python3.10/site-packages/typing_extensions.py", line 91, in _check_generic
    raise TypeError(f"{cls} is not a generic class")
TypeError: <class 'homeassistant.components.nextdns.NextDnsUpdateCoordinator'> is not a generic class
```

Reply:
Considering the complexity of making Generic classes, it might be worth making a preliminary PR for all the Generics, and then come back to this one afterwards.

Reply:
Sample implementation in #74517